### PR TITLE
HAAR-3304: Ensure correct field is used for prison id for restricted patients template

### PR DESCRIPTION
**Pull request by indyv-moj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/services/GeneratePdfServiceRestrictedPatientsTest.kt [hunk 1]
```diff
@@ -41,16 +43,19 @@ class GeneratePdfServiceRestrictedPatientsTest {
     val testInput = mapOf(
       "prisonerNumber" to "A1234AA",
       "supportingPrisonDescription" to "HMP Exeter",
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/services/GeneratePdfServiceRestrictedPatientsTest.kt:45]: I'm going to remove `supportingPrisonDescription` from restricted patients so worth removing this from here too

File: src/test/resources/pdf/testutil/stubs/hmpps-restricted-patients-api-stub.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
   "content": {
     "prisonerNumber": "A1234AA",
     "supportingPrisonDescription": "HMP Exeter",
```
Comment (petergphillips) [inline src/test/resources/pdf/testutil/stubs/hmpps-restricted-patients-api-stub.json:4]: again would be good to remove use of description here so I can remove from restricted patients api

File: src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/services/GeneratePdfServiceRestrictedPatientsTest.kt (hunk 1 above)
Comment (indyv-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/services/GeneratePdfServiceRestrictedPatientsTest.kt]: Done

File: src/test/resources/pdf/testutil/stubs/hmpps-restricted-patients-api-stub.json (hunk 2 above)
Comment (indyv-moj) [reply src/test/resources/pdf/testutil/stubs/hmpps-restricted-patients-api-stub.json]: Removed

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**